subsys/bluetooth/mesh/net.c

```diff
@@ -285,27 +287,27 @@ bool bt_mesh_net_iv_update(uint32_t iv_index, bool iv_update)
 		return false;
 	}
 
-	if ((iv_index > bt_mesh.iv_index + 1) ||
-	    (iv_index == bt_mesh.iv_index + 1 &&
-	     (atomic_test_bit(bt_mesh.flags, BT_MESH_IVU_IN_PROGRESS) || !iv_update))) {
+	iv_update_is_same = (atomic_test_bit(bt_mesh.flags, BT_MESH_IVU_IN_PROGRESS) == iv_update);
+
+	/* MshPRTv1.1 allows to initiate an
+	 * IV Index Recovery procedure if previous IV update has
+	 * been missed. This allows the node to remain
+	 * functional.
+	 *
+	 * Upon receiving and successfully authenticating a
+	 * Secure Network beacon for a primary subnet whose
+	 * IV Index is 1 or more higher than the current known IV
+	 * Index, the node shall set its current IV Index and its
+	 * current IV Update procedure state from the values in
+	 * this Secure Network beacon.
+	 */
+	if (((iv_index - bt_mesh.iv_index) + iv_update_is_same) > 1) {
```

Comment:
This logic seems really odd. A boolean result is used in mathematical expression to decide whether to proceed or not. This does not make it easy to infer the logic unless someone really reads the multiple lines. 

I will not prefer to change anything here unless there is any real bug. Current implementation is more readable IMO.